### PR TITLE
Fixed `ReflectionParameter::allowsNull()` for union types

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/ReflectionParameter.php [hunk 1]
```diff
@@ -308,15 +307,18 @@ public function allowsNull(): bool
             return true;
         }
 
-        if ($this->node->type instanceof NullableType) {
-            return true;
-        }
+        $type = $this->getType();
```
Comment (Ocramius) [inline src/Reflection/ReflectionParameter.php]: Perhaps we can simplify this further:

```php
$type = $this->getType();

if (null === $type) {
    return true;
}

// ...
```

That removes a potentially pricey call, and removes `assert()`

**Review by Ocramius — APPROVED** [review]
Good simplification: up to you if I should merge as-is, or if the suggested minor improvement should be added first

File: src/Reflection/ReflectionParameter.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionParameter.php]: Wil do in 30 minutes.

Comment (kukulich) [reply src/Reflection/ReflectionParameter.php]: Done.

**Review by asgrim — APPROVED** [review]

**Pull request merged**